docs/user_guide/structure.md

```diff
@@ -0,0 +1,199 @@
+```--producers```. The histograms are saved in a pickle file.
+TODO: more informations?
+
+(PlotVariablesTasks)=
+- ```PlotVariables*```, ```PlotShiftedVariables*``` (e.g.
+{py:class}`~columnflow.tasks.plotting.PlotVariables1D`,
+{py:class}`~columnflow.tasks.plotting.PlotVariables2D`,
+{py:class}`~columnflow.tasks.plotting.PlotShiftedVariables1D`): Tasks to plot the histograms created by
+{py:class}`~columnflow.tasks.histograms.CreateHistograms` using the python package
+[matplotlib](https://matplotlib.org/) with [mplhep](https://mplhep.readthedocs.io/en/latest/) style.
+Several plot types are possible, including
+plots of variables for different physical processes or plots of variables for a single physical
+process but different shifts (e.g. jet-energy correction variations). The argument ```--variables```
+followed by the name of the variables defined in the analysis config, separated by a comma, is
+needed for these tasks to run. It is also possible to replace the ```--datasets``` argument
+for these tasks by the ```--processes``` argument followed by the name of the physical processes to
+be plotted, as defined in the analysis config. For the ```PlotShiftedVariables*```plots, the
+argument ```shift-sources``` is needed and replaces the argument ```shift```. The output format for
+these plots can be given with the ```--file-types``` argument. It is possible to set a default for the
+variables in the analysis config.
+
+- {py:class}`~columnflow.tasks.cms.inference.CreateDatacards`: TODO
+
+- ```PlotCutflow*``` (e.g. {py:class}`~columnflow.tasks.cutflow.PlotCutflow`,
+{py:class}`~columnflow.tasks.cutflow.PlotCutflowVariables1D`): Tasks to plot the histograms created
+by {py:class}`~columnflow.tasks.cutflow.CreateCutflowHistograms`. The
+{py:class}`~columnflow.tasks.cutflow.PlotCutflowVariables1D` are plotted in a similar way to the
+["PlotVariables*"](PlotVariablesTasks) tasks. The difference is that these plots show the selection
+yields of the different selection steps defined in
+{py:class}`~columnflow.tasks.selection.SelectEvents` instead of only after the
+{py:class}`~columnflow.tasks.reduction.ReduceEvents` procedure. The selection steps to be shown
+can be chosen with the ```--selector-steps``` argument. Without further argument, the outputs are
+as much plots as the number of selector steps given. On the other hand, the
+{py:class}`~columnflow.tasks.cutflow.PlotCutflow` task gives a single histograms containing only
+the total event yields for each selection step given.
+
+- ```Merge``` tasks (e.g. {py:class}`~columnflow.tasks.reduction.MergeReducedEvents`,
+{py:class}`~columnflow.tasks.histograms.MergeHistograms`): Tasks to merge the local outputs from
+the various occurences of the corresponding tasks. TODO: details? why needed? Only convenience?
+Or I/O?
+
+- {py:class}`~columnflow.tasks.cms.external.CreatePileUpWeights`: TODO
+
+maybe TODO: each task separately?
```

Comment:
I agree with you that we should maybe have multiple sections for the tasks. For me, it feels like this section frontloads lots of information the user might not really need at such an early point in the guide. I think it maybe makes more sense to only introduce the tasks and their purpose here without mentioning all of the details of parameters and CSPs etc. but instead link to some section which describes more in-depth how to actually work with this task.

Maybe a full `tasks` section (similar to the `building_blocks` section) would be better, e.g. with sub-sections for:
- CalibrateEvents
- SelectEvents
- ReduceEvents and the following merging pipeline
- ProduceColumns
- The full Machine Learning pipeline
- Creating and Merging of Histograms
- Plotting tasks
- Cutflow tasks

Of course, we would have to make sure that we do not replicate information that is present in the `building_blocks` section and vice versa.

Reply:
As of now I have the general introduction of the purpose of the task, its specific parameters (the ones created specifically for this task), the output of the task and additional general infos that might be needed for new users (e.g. keep_columns for ReduceEvents), the inner working with the corresponding objects and examples are in the building blocks section, e.g. the "Running the SelectEvents task" in the "Selections" part of the building blocks... Should this be changed? If so, which parts are to be dropped in the general introduction?

Reply:
for the future